pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -796,16 +796,21 @@ public class Utils {
         Set<String> previousTriggers = new HashSet<>()
         Set<String> previousParameters = new HashSet<>()
 
-        DeclarativeJobPropertyTrackerAction previousAction = null
-        WorkflowRun previousBuild = r.getPreviousBuild()
-        if (previousBuild != null) {
-            previousAction = previousBuild.getAction(DeclarativeJobPropertyTrackerAction.class)
-            if (previousAction != null) {
-                previousProperties.addAll(previousAction.getJobProperties())
-                previousTriggers.addAll(previousAction.getTriggers())
-                previousParameters.addAll(previousAction.getParameters())
+        // First, use the action from the job if it's present.
+        DeclarativeJobPropertyTrackerAction previousAction = j.getAction(DeclarativeJobPropertyTrackerAction.class)
```

Comment:
The `Trigger`s and `ParameterDefinition`s are defined via their `JobProperty`s, yes, but I need to be able to say "Ok, which `Trigger`s were defined in the `Jenkinsfile` vs which ones were defined in the UI", which is a level of granularity that doesn't function with `properties`. I couldn't find a way to do that with just `JobPropertyTrackerAction`, and I don't feel like it makes sense to introspect the properties passed to `properties` for `PipelineTriggersJobProperty` and `ParametersDefinitionProperty` and their contents, etc...